### PR TITLE
Add validation handler and error encoder

### PR DESCRIPTION
**Pull request by codyaray**
Implement https://github.com/getkin/kin-openapi/issues/194 and https://github.com/getkin/kin-openapi/issues/195

This is what I had to add on top of kin-openapi in order to provide openapi validation errors in my API service. More discussion at https://github.com/getkin/kin-openapi/issues/194#issuecomment-605488953

### CODE REVIEW

File: openapi3filter/validation_handler.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	return nil
+}
+
+func (h *ValidationHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	err := h.validateRequest(r)
+	if err != nil {
+		h.ErrorEncoder(r.Context(), err, w)
+		return
+	}
+	// TODO: validateResponse
```
Comment (fenollp) [inline openapi3filter/validation_handler.go:50]: Indeed

File: openapi3filter/validation_kit.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
+package openapi3filter
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+)
+
+///////////////////////////////////////////////////////////////////////////////////
+// We didn't want to tie kin-openapi too tightly with go-kit.
```
Comment (fenollp) [inline openapi3filter/validation_kit.go:10]: Let's not copy https://github.com/go-kit/kit/blob/266ff8dc37c693d0649707e519c93c1f85868bdc/transport/http/server.go here but rather import this package.

Comment (fenollp) [reply openapi3filter/validation_kit.go:10]: On second thought this brings a pretty large dependency tree so disregard the above comment.

File: openapi3filter/validation_handler.go (hunk 1 above)
Comment (fenollp) [reply openapi3filter/validation_handler.go:50]: This can come later as another PR.

**Comment by fenollp** [comment]
Could you look at the WIP new router at https://github.com/getkin/kin-openapi/pull/210?
I believe I could add a `Handle` or `HandleFunc` that could help you there.

**Comment by wolfeidau** [comment]
This looks pretty close, anything i can do to help?

**Comment by codyaray** [comment]
@fenollp thanks for merging. I agree that I really don't like matching on error strings. That was the only option given what we have at the moment though.

Sorry I missed your follow-up questions. What did you mean by "producing correct JSON pointers" ? I assume you mean this kinda stuff? https://github.com/getkin/kin-openapi/pull/197/files#diff-ee2d579818e5124734fb3ce2cd7e87acR155

**Pull request merged**